[WIP][Drup-354] Display plans.

Jaesin opened the pull request:
This work also contains the rate plan entity.

Jaesin committed:
[DRUP-402] Add the rate plan entity.

Jaesin committed:
[DRUP-402] Missing product controller.

Jaesin committed:
[DRUP-402] Some cleanup and theme implementation.

Jaesin committed:
[DRUP-402][DRUP-354] Display plans within monetization packages.

cnovak commented on apigee_m10n.links.menu.yml:
```diff
@@ -14,3 +14,10 @@ apigee_m10n.packages:
   menu_name: account
   weight: -20
+apigee_m10n.rate_plan:
+  title: 'Rate Plans'
```
```suggestion
  title: 'Rate plans'
```

cnovak reviewed (commented):
When going to user/1/monetization/packages URL, I am getting:
```
Error: Call to a member function add() on null in Drupal\apigee_m10n\Entity\RatePlan->set() (line 121 of modules/contrib/apigee_m10n/src/Entity/FieldableMonetizationEntityBaseTrait.php).

Drupal\apigee_m10n\Entity\RatePlan->set('keepOriginalStartDate', 1) (Line: 189)
Drupal\apigee_m10n\Entity\Storage\FieldableMonetizationEntityStorageBase->initFieldValues(Object, Array) (Line: 151)
Drupal\apigee_m10n\Entity\Storage\FieldableMonetizationEntityStorageBase->doCreate(Array) (Line: 221)
...
```

mxr576 commented on src/Entity/FieldableMonetizationEntityBaseTrait.php:
```diff
@@ -0,0 +1,128 @@
+  }
+
+  /**
+   * This is mostly copied from `FieldableEdgeEntityBaseTrait` but we are adding
```
Could you add this feature without code duplication? :skull_and_crossbones:
For example by introducing a base class for mint entities that extends the inherited set() method from this trait?

Jaesin replied:
There have been some recent changes to `apigee_edge` that breaks what we are doing here. We will see what we can to about this when we pull in those changes and make the necessary fixes.

mxr576 commented on src/Entity/Storage/MonetizationEntityStorageBase.php:
```diff
@@ -0,0 +1,337 @@
+use Psr\Log\LoggerInterface;
+use Symfony\Component\DependencyInjection\ContainerInterface;
+
+abstract class MonetizationEntityStorageBase extends EntityStorageBase {
```
This is also a copy-paste clone :skull_and_crossbones:  of https://github.com/apigee/apigee-edge-drupal/blob/8.x-1.x/src/Entity/Storage/EdgeEntityStorageBase.php. Why did you extended the EdgeEntityStorageBase class instead?

Jaesin committed:
[DRUP-402] Fix API package formatter target.

Jaesin committed:
[DRUP-402] Don’t try to use attributes.

Jaesin committed:
[DRUP-402] Show rate plan details ID until the formatter is built out.

Commit:
[DRUP-354]  Rate plan detail theme template.

Commit:
[DRUP-354]  Rate plan detail CSS.

cnovak committed:
Update apigee_m10n.links.menu.yml

Co-Authored-By: Jaesin <Jaesin@users.noreply.github.com>

Jaesin committed:
[DRUP-402] The  currency formatter should only apply to a `apigee_currency` field.

Commit:
[DRUP-354]  Rate plan detail - adding more variables.

Commit:
[DRUP-354]  Rate plan detail - Human readable labels.

Commit:
[DRUP-354]  Rate plan detail - Use Drupal field styles.

Commit:
[DRUP-354]  Rate Plan Details - Removing CSS styles.

Commit:
[DRUP-354]  Rate Plan Details - Render as table and use only variables listed on D7.

cnovak commented:
@mxr576 I created issues to fix the MonetizationEntityStorageBase.php and FieldableMonetizationEntityBaseTrait.php as you commented above.

https://apigeesc.atlassian.net/browse/DRUP-424
https://apigeesc.atlassian.net/browse/DRUP-425

Jaesin committed:
[DRUP-402] Drop some static caching because performance didn’t test as beneficial.

zakiya commented:
I signed the CLA.

Jaesin committed:
[DRUP-354] Display package details by default.

Jaesin committed:
Getting credentials on construct is causing errors in testing.

Jaesin committed:
[DRUP-437] Remove type hinting requirement for functions.

Core doesn't conform to this standard.
@see: https://www.drupal.org/project/coder/issues/2803251
@see: https://www.drupal.org/project/drupal/issues/1158720

Jaesin committed:
[DRUP-437] PHPCS fixes.

Jaesin committed:
[DRUP-437] Fix the test setup base after rebase.

Jaesin committed:
[DRUP-437] Fix static accessor after rebase.

Jaesin committed:
[DRUP-437] Formatting fix.
